### ISSUE TITLE
Error running with doom emacs

### ISSUE THREAD

**Issue by BlueDrink9** [jcs-elpa/company-fuzzy#18]
When enabling this minor mode in a buffer in Doom emacs, I get this message when company tries to pull up suggestions:

```
Company: frontend company-pseudo-tooltip-frontend error "Company: backend company-fuzzy-all-other-backends error "Wrong type argument: numberp, nil" with args (pre-render config_data nil)" on command post-command
```

Let me know what else you want me to do to help diagnose, I am a little new to emacs

**Comment by BlueDrink9** [comment]
Still thinking on this, I just haven't figured out how to do a `-Q` that still loads doom packages

**Comment by BlueDrink9** [comment]
In the meanwhile, this is a backtrace from a doom install without my config.

```
Debugger entered--Lisp error: (wrong-type-argument numberp nil)
  nth(nil (#("dec" 0 3 (yas-annotation "def decorator" yas-template #s(yas--template :key "dec" :content "def ${1:decorator}(func):\n    $2\n    def _$1(*args..." :name "def decorator" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "dec" :menu-binding-pair ((menu-item "def decorator" (lambda nil ... ...) :keys "dec =>")) :group ("definitions") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) #("def" 0 3 (yas-annotation "def function" yas-template #s(yas--template :key "def" :content "def ${1:func_name}($2):\n    ${3:`(or % \"pass\")`}" :name "def function" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "def" :menu-binding-pair ((menu-item "def function" (lambda nil ... ...) :keys "def =>")) :group ("definitions") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) #("defc" 0 4 (yas-annotation "classmethod" yas-template #s(yas--template :key "defc" :content "@classmethod\ndef ${1:method_name}(cls, $1):\n    $0" :name "classmethod" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "defc" :menu-binding-pair ((menu-item "classmethod" (lambda nil ... ...) :keys "defc =>")) :group ("object oriented") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) #("defd" 0 4 (yas-annotation "method_docstring" yas-template #s(yas--template :key "defd" :content "def ${1:name}(self$2):\n    \\\"\\\"\\\"$3\n    ${2:$(pyth..." :name "method_docstring" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "defd" :menu-binding-pair ((menu-item "method_docstring" (lambda nil ... ...) :keys "defd =>")) :group ("object oriented") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) #("defm" 0 4 (yas-annotation "method" yas-template #s(yas--template :key "defm" :content "def ${1:method_name}(self${2:, $3}):\n    ${4:`(or ..." :name "method" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "defm" :menu-binding-pair ((menu-item "method" (lambda nil ... ...) :keys "defm =>")) :group ("object oriented") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) #("defs" 0 4 (yas-annotation "static" yas-template #s(yas--template :key "defs" :content "@staticmethod\ndef ${1:method_name}($1):\n    $0" :name "static" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "defs" :menu-binding-pair ((menu-item "static" (lambda nil ... ...) :keys "defs =>")) :group nil :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2))))
  (let* ((str-len (length str)) (prefix (or (company-fuzzy--backend-prefix-candidate str 'match) "")) (cur-selection (nth company-selection company-candidates)) (splitted-section (remove "" (split-string str " "))) (process-selection (nth 0 splitted-section)) (selected (string= cur-selection process-selection)) (selected-face (if selected 'company-tooltip-common-selection 'company-tooltip-common)) (selected-common-face (if selected 'company-tooltip-selection 'company-tooltip)) (splitted-c (remove "" (split-string prefix "")))) (set-text-properties 0 str-len nil str) (font-lock-prepend-text-property 0 str-len 'face selected-common-face str) (let ((--dolist-tail-- splitted-c)) (while --dolist-tail-- (let ((c (car --dolist-tail--))) (let ((pos (company-fuzzy--string-match-p ... str))) (while (and (numberp pos) (< pos str-len)) (font-lock-prepend-text-property pos (1+ pos) 'face selected-face str) (setq pos (company-fuzzy--string-match-p ... str ...)))) (setq --dolist-tail-- (cdr --dolist-tail--))))))
  (if annotation-p nil (let* ((str-len (length str)) (prefix (or (company-fuzzy--backend-prefix-candidate str 'match) "")) (cur-selection (nth company-selection company-candidates)) (splitted-section (remove "" (split-string str " "))) (process-selection (nth 0 splitted-section)) (selected (string= cur-selection process-selection)) (selected-face (if selected 'company-tooltip-common-selection 'company-tooltip-common)) (selected-common-face (if selected 'company-tooltip-selection 'company-tooltip)) (splitted-c (remove "" (split-string prefix "")))) (set-text-properties 0 str-len nil str) (font-lock-prepend-text-property 0 str-len 'face selected-common-face str) (let ((--dolist-tail-- splitted-c)) (while --dolist-tail-- (let ((c (car --dolist-tail--))) (let ((pos ...)) (while (and ... ...) (font-lock-prepend-text-property pos ... ... selected-face str) (setq pos ...))) (setq --dolist-tail-- (cdr --dolist-tail--)))))))
  company-fuzzy--pre-render(#("dec" 0 3 (yas-annotation "def decorator" yas-template #s(yas--template :key "dec" :content "def ${1:decorator}(func):\n    $2\n    def _$1(*args..." :name "def decorator" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "dec" :menu-binding-pair ((menu-item "def decorator" (lambda nil (interactive) (yas--expand-or-visit-from-menu ... "dec")) :keys "dec =>")) :group ("definitions") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) nil)
  (cond ((eql command 'interactive) (company-begin-backend 'company-fuzzy-all-other-backends)) ((eql command 'prefix) (company-fuzzy--get-prefix)) ((eql command 'annotation) (company-fuzzy--extract-annotation arg)) ((eql command 'candidates) (company-fuzzy-all-candidates)) ((eql command 'pre-render) (company-fuzzy--pre-render arg (nth 0 ignored))) ((memql command '(or doc-buffer kind)) (company-fuzzy--backend-command arg command)))
  company-fuzzy-all-other-backends(pre-render #("dec" 0 3 (yas-annotation "def decorator" yas-template #s(yas--template :key "dec" :content "def ${1:decorator}(func):\n    $2\n    def _$1(*args..." :name "def decorator" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "dec" :menu-binding-pair ((menu-item "def decorator" (lambda nil (interactive) (yas--expand-or-visit-from-menu ... "dec")) :keys "dec =>")) :group ("definitions") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) nil)
  apply(company-fuzzy-all-other-backends (pre-render #("dec" 0 3 (yas-annotation "def decorator" yas-template #s(yas--template :key "dec" :content "def ${1:decorator}(func):\n    $2\n    def _$1(*args..." :name "def decorator" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "dec" :menu-binding-pair ((menu-item "def decorator" (lambda nil ... ...) :keys "dec =>")) :group ("definitions") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) nil))
  company-call-backend-raw(pre-render #("dec" 0 3 (yas-annotation "def decorator" yas-template #s(yas--template :key "dec" :content "def ${1:decorator}(func):\n    $2\n    def _$1(*args..." :name "def decorator" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "dec" :menu-binding-pair ((menu-item "def decorator" (lambda nil (interactive) (yas--expand-or-visit-from-menu ... "dec")) :keys "dec =>")) :group ("definitions") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) nil)
  apply(company-call-backend-raw (pre-render #("dec" 0 3 (yas-annotation "def decorator" yas-template #s(yas--template :key "dec" :content "def ${1:decorator}(func):\n    $2\n    def _$1(*args..." :name "def decorator" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "dec" :menu-binding-pair ((menu-item "def decorator" (lambda nil ... ...) :keys "dec =>")) :group ("definitions") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) nil))
  company--force-sync(company-call-backend-raw (pre-render #("dec" 0 3 (yas-annotation "def decorator" yas-template #s(yas--template :key "dec" :content "def ${1:decorator}(func):\n    $2\n    def _$1(*args..." :name "def decorator" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "dec" :menu-binding-pair ((menu-item "def decorator" (lambda nil ... ...) :keys "dec =>")) :group ("definitions") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) nil) company-fuzzy-all-other-backends)
  company-call-backend(pre-render #("dec" 0 3 (yas-annotation "def decorator" yas-template #s(yas--template :key "dec" :content "def ${1:decorator}(func):\n    $2\n    def _$1(*args..." :name "def decorator" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "dec" :menu-binding-pair ((menu-item "def decorator" (lambda nil (interactive) (yas--expand-or-visit-from-menu ... "dec")) :keys "dec =>")) :group ("definitions") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) nil)
  company--pre-render(#("dec" 0 3 (yas-annotation "def decorator" yas-template #s(yas--template :key "dec" :content "def ${1:decorator}(func):\n    $2\n    def _$1(*args..." :name "def decorator" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "dec" :menu-binding-pair ((menu-item "def decorator" (lambda nil (interactive) (yas--expand-or-visit-from-menu ... "dec")) :keys "dec =>")) :group ("definitions") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)))
  company-fill-propertize(#("dec" 0 3 (yas-annotation "def decorator" yas-template #s(yas--template :key "dec" :content "def ${1:decorator}(func):\n    $2\n    def _$1(*args..." :name "def decorator" :condition nil :expand-env nil :load-file "/home/user/.emacs.d/.local/straight/build-27.2/..." :save-file nil :keybinding nil :uuid "dec" :menu-binding-pair ((menu-item "def decorator" (lambda nil (interactive) (yas--expand-or-visit-from-menu ... "dec")) :keys "dec =>")) :group ("definitions") :perm-group nil :table #s(yas--table :name "python-mode" :hash #<hash-table equal 102/145 0x155bef5fdd69> :uuidhash #<hash-table equal 146/217 0x155bef5fdf15> :parents nil :direct-keymap (keymap))) yas-prefix-offset 2)) #("def decorator <yasnippet>" 13 25 (face company-fuzzy-annotation-face)) 33 nil #("  " 0 1 (display (image :file "/home/user/.emacs.d/.local/straight/build-27.2/..." :type svg :width 16 :height 16 :ascent center :background nil)) 1 2 (display (space :width (2)))) " ")
  company--create-lines(nil 7)
  company-pseudo-tooltip-show(40 12 nil)
  company-pseudo-tooltip-show-at-point(50572 2)
  company-pseudo-tooltip-frontend(post-command)
  company-call-frontends(post-command)
  company-post-command()
  company-idle-begin(#<buffer interface.py> #<window 3 on interface.py> 19509 50572)
  apply(company-idle-begin (#<buffer interface.py> #<window 3 on interface.py> 19509 50572))
  timer-event-handler([t 24885 52817 502103 nil company-idle-begin (#<buffer interface.py> #<window 3 on interface.py> 19509 50572) nil 936000])
```

**Comment by BlueDrink9** [comment]
Managed to reproduce with an `emacs -Q`. Thank you for your patience. I'm using straight.el for easier package setup, I hope that's ok.

```elisp
(load-file "~/.emacs.d/straight/repos/straight.el/bootstrap.el")
(straight-use-package 'company)
(straight-use-package 'company-fuzzy)
(company-mode)
(company-tng-mode)
(company-fuzzy-mode)
```

Message with that combination:


```
Company: An error occurred in post-command
Company: frontend company-pseudo-tooltip-frontend error "Company: backend company-fuzzy-all-other-backends error "Wrong type argument: numberp, nil" with args (pre-render combine-after-change-calls nil)" on command post-command
```

**Comment by jcs090218** [comment]
Thanks for the detail report!

I have applied a fix in 569ff8d0897da2832b6a5086c321ab9cf4263b17.
